core/src/main/java/io/crate/core/collections/PrependedRow.java

```diff
@@ -27,14 +27,17 @@
  */
 public class PrependedRow implements Row {
 
-    private final Row delegateRow;
     private final byte inputId;
+    private Row delegateRow;
 
-    public PrependedRow(Row delegateRow, byte inputId) {
-        this.delegateRow = delegateRow;
+    public PrependedRow(byte inputId) {
```

Comment:
think this could use a comment that `setDelegate` needs to be called, otherwise this will result in NPEs